### PR TITLE
[berdy] Modified floating base Berdy sensor matrix to include joint t…

### PR DESCRIPTION
**Pull request by prashanthr05**
…orque measurements

cc @traversaro

### CODE REVIEW

File: src/estimation/src/BerdyHelper.cpp [hunk 1]
```diff
@@ -1244,16 +1244,49 @@ bool BerdyHelper::computeBerdySensorMatrices(SparseMatrix<iDynTree::ColumnMajor>
...
 
             matrixYElements.pushTriplet(Triplet(sensorRange.offset, jointTrqRange.offset, 1));
 
             // bY for the joint torques is zero
+          }
+        }
+        else 
+        {
+          assert(false);
+          // It should be something like a submatrix with S^T (see lines 907)
```
Comment (traversaro) [inline src/estimation/src/BerdyHelper.cpp]: Leftover

File: src/estimation/src/BerdyHelper.cpp [hunk 2]
```diff
@@ -1244,16 +1244,49 @@ bool BerdyHelper::computeBerdySensorMatrices(SparseMatrix<iDynTree::ColumnMajor>
...
             IndexRange jointTrqRange = this->getRangeDOFVariable(DOF_TORQUE,idx);
 
             matrixYElements.pushTriplet(Triplet(sensorRange.offset, jointTrqRange.offset, 1));
 
             // bY for the joint torques is zero
+          }
+        }
+        else 
+        {
+          assert(false);
```
Comment (traversaro) [inline src/estimation/src/BerdyHelper.cpp]: I guess you are still not compiling in debug. : )

File: src/estimation/src/BerdyHelper.cpp [hunk 3]
```diff
@@ -1244,16 +1244,49 @@ bool BerdyHelper::computeBerdySensorMatrices(SparseMatrix<iDynTree::ColumnMajor>
...
+          // It should be something like a submatrix with S^T (see lines 907)
+          // We have to map the joint wrench to the joint torques, since joint wrench is considered as dynamic variable
+          for (TraversalIndex traversalEl = 1;
+               traversalEl < static_cast<TraversalIndex>(m_dynamicsTraversal.getNrOfVisitedLinks());
+               traversalEl++)
+          {
+            LinkConstPtr visitedLink = m_dynamicsTraversal.getLink(traversalEl);
```
Comment (traversaro) [inline src/estimation/src/BerdyHelper.cpp]: 4 spaces for indentation consistency with the rest of the code, thanks!

File: src/estimation/src/BerdyHelper.cpp (hunk 2 above)
Comment (prashanthr05) [reply src/estimation/src/BerdyHelper.cpp]: I am changing this to ```assert(m_options.berdyVariant == BERDY_FLOATING_BASE);```

**Comment by prashanthr05** [comment]
cc @traversaro  made the required changes in the latest commit

**Comment by traversaro** [comment]
cc @claudia-lat @lucaTagliapietra Without this fix, the `includeAllJointTorquesAsSensors` option was broken when used with `BERDY_FLOATING_BASE`.

**Pull request merged**